hadoop-hdds/rocksdb-checkpoint-differ/src/main/java/org/apache/ozone/rocksdiff/RocksDBCheckpointDiffer.java

```diff
@@ -695,17 +877,45 @@ public synchronized void printMutableGraph(
     }
   }
 
+  private String getCompactionLogFilenameGivenCpPath(String cpPath) {
+    return cpPath + "-" + CURRENT_COMPACTION_LOG_FILENAME;
+  }
 
   public void createSnapshot(RocksDB rocksDB) throws InterruptedException {
 
     LOG.warn("Current time is::" + System.currentTimeMillis());
     long t1 = System.currentTimeMillis();
 
     cpPath = cpPath + lastSnapshotCounter;
+    // Delete the checkpoint dir if it already exists
+    File dir = new File(cpPath);
+    if (dir.exists()) {
+      deleteDirectory(dir);  // TODO: FOR EASE OF TESTING ONLY. DO NOT DELETE DIR WHEN MERGING
+    }
+
     createCheckPoint(rocksDbPath, cpPath, rocksDB);
     allSnapshots[lastSnapshotCounter] = new Snapshot(cpPath,
     lastSnapshotPrefix, lastSnapshotCounter);
 
+    // Rename compactions.log
+    final String existingFilename = CURRENT_COMPACTION_LOG_FILENAME;
+    final String newFilename = getCompactionLogFilenameGivenCpPath(cpPath);
+    try {
+
+      // Dev: Remove target is already exists
+      if (new File(newFilename).exists()) {
+        new File(newFilename).delete();
+      }
+
+      if (new File(existingFilename).exists()) {
+        Files.move(Paths.get(existingFilename), Paths.get(newFilename));
```

Comment:
Does there need to be synchronization with the OnCompactionComplete() listener, to ensure this file is consistent at the time of the move?

Reply:
Ah, I see you've added pauseBackgroundWork() in SnapshotManager.  I wonder how long that might make it pause for?

Reply:
Hi George, we realized we actually couldn't `pauseBackgroundWork` because RDB checkpointing also counts as a kind of background work. After `pauseBackgroundWork()` is called, the RDB checkpointing call would wait indefinitely. So we switched to using a new compaction log file name every time a new snapshot is created in #3824 .